File: book_client/src/app/model/Book.ts

export class Book {
    id: number;
    name: string;
    author: string;
    url: String;


    constructor(bookid: number, bookname: string, bookauthor: string, url: String) {
        this.id = bookid;
        this.name = bookname;
        this.author = bookauthor;
        this.url = url;

    }

    getBook(): string {
        return 'bookid:' + this.id + ","
            + 'bookname:' + this.name + ","
            + 'bookurl:' + this.url + ","
            + 'bookauthor:' + this.author;
    }
}

File: server/src/main/java/com/project/server/jpa/entity/Book.java
package com.project.server.jpa.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Table;
@Entity
@Table(name = "book")
public class Book {
    @Id
    @GeneratedValue
    private Long id;

    @Column(nullable = false, name = "bookName")
    private String name;

    @Column(name = "bookAuthor")
    private  String author;

    @Override
    public String toString() {
        return "Book{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", author='" + author + '\'' +
                ", url='" + url + '\'' +
                '}';
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    @Column(name = "url")
    private String url;
    
    public Book(String name, String author){
        this.name = name;
        this.author = author;
    }

    public  Book(){}
    public String getAuthor() {
        return author;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    
    public void setId(Long id) {
        this.id = id;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

}
